src/updateFileFromServer.ts

```diff
@@ -118,45 +120,60 @@ export async function toggleServerTaskStatus(e: Editor, settings: TodoistSetting
 	}
 	catch (e){
 		console.log("todoist text error: ", e);
-		new Notice("Todoist Text: Error trying to update task status. See console log for more details.")
+		new Notice("Todoist Text: Error trying to update task status. See console log for more details.");
 	}
 }
 
-async function getServerData(todoistQuery: string, authToken: string, showSubtasks: boolean): Promise<string> {
-	const api = new TodoistApi(authToken)
-
+async function getServerData(todoistQuery: string, authToken: string, showSubtasks: boolean, showPriority: boolean, showLink: boolean, showDescription: boolean, noDateSubtasks: boolean, todaysSubtasks: boolean): Promise<string> {
+	const api = new TodoistApi(authToken);
 	const tasks = await callTasksApi(api, todoistQuery);
-	
+
 	if (tasks.length === 0){
 		new Notice(`Todoist text: You have no tasks matching filter "${todoistQuery}"`);
 	}
 	
 	let returnString = "";
 	if (showSubtasks) {
-		// work through all the parent tasks
-		let parentTasks = tasks.filter(task => task.parentId == null);
-		parentTasks.forEach(task => {
-			returnString = returnString.concat(getFormattedTaskDetail(task, 0, false));
-			returnString = returnString.concat(getSubTasks(tasks, task.id, 1));
-		})
+		let subtaskQuery = "subtask"; // default pull query of all subtasks
+		if (todaysSubtasks && noDateSubtasks) {
+			subtaskQuery = "subtask & due: today | subtask & no due date"; // excludes subtasks with due dates that are not set to today
+		} else {
+			if (noDateSubtasks) {
+				subtaskQuery = "subtask"; // default pull query of all subtasks, regardless of due date
+			} else if (todaysSubtasks && !noDateSubtasks) {
+				subtaskQuery = "subtask & due: today"; // limits to only subtasks with due date of today
+			} else {
```

Comment:
i think the nesting here gets a bit harder to follow. if you could flatten the `if`s, that'd be nice